src/server/scripts/Spells/spell_shaman.cpp

```diff
@@ -352,6 +355,58 @@ class spell_sha_crash_lightning : public SpellScript
     size_t _targetsHit = 0;
 };
 
+// 378270 - Deeply Rooted Elements
+class spell_sha_deeply_rooted_elements : public AuraScript
+{
+    PrepareAuraScript(spell_sha_deeply_rooted_elements);
+
+    bool Validate(SpellInfo const* spellInfo) override
+    {
+        return ValidateSpellInfo({ SPELL_SHAMAN_LAVA_BURST, SPELL_SHAMAN_ASCENDANCE_ELEMENTAL, SPELL_SHAMAN_ASCENDANCE_ENHANCEMENT, SPELL_SHAMAN_ASCENDANCE_RESTORATION })
+            && ValidateSpellEffect({ { spellInfo->Id, EFFECT_0 } });
+    }
+
+    bool CheckProc(AuraEffect const* /*aurEff*/, ProcEventInfo& procInfo)
+    {
+        if (procInfo.GetSpellInfo()->Id == SPELL_SHAMAN_LAVA_BURST && procInfo.GetActor()->ToPlayer()->GetPrimarySpecialization() != TALENT_SPEC_SHAMAN_ELEMENTAL)
```

Comment:
This is still wrong, you're only checking for elemental shaman.

```suggestion
            if (player->GetPrimarySpecialization() == ChrSpecialization::ShamanElemental && eventInfo.GetProcSpell()->GetSpellInfo()->Id == SPELL_SHAMAN_LAVA_BURST)
        return roll_chance_i(GetEffectInfo(EFFECT_1).CalcValue());
    else if (player->GetPrimarySpecialization() == ChrSpecialization::ShamanEnhancement && eventInfo.GetProcSpell()->GetSpellInfo()->Id == SPELL_SHAMAN_STORMSTRIKE)
        return roll_chance_i(GetEffectInfo(EFFECT_2).CalcValue());
    else if (player->GetPrimarySpecialization() == ChrSpecialization::ShamanRestoration && eventInfo.GetProcSpell()->GetSpellInfo()->Id == SPELL_SHAMAN_RIPTIDE)
        return roll_chance_i(GetEffectInfo(EFFECT_3).CalcValue());
```